xarray/core/accessor_str.py

```diff
@@ -271,13 +271,13 @@ def _re_compile(
         if getattr(pat, "dtype", None) != np.object_:
             pat = self._stringify(pat)
         func = lambda x: re.compile(x, flags=flags)
-        if isinstance(pat, np.ndarray):
-            # apply_ufunc doesn't work for numpy arrays with output object dtypes
-            func = np.vectorize(func)
-            return func(pat)
-        else:
+        if not isinstance(pat, np.ndarray):
             return _apply_str_ufunc(func=func, obj=pat, dtype=np.object_)
 
+        # apply_ufunc doesn't work for numpy arrays with output object dtypes
+        func = np.vectorize(func)
+        return func(pat)
```

Comment:
I also don't think this is an improvement. Early `return` and one less level of indentation is help when there's a large block of code that doesn't need to be indented, but here it's only three short lines. It is easier to follow the control flow (seeing the mutually exclusive options) when the alternative is in an `else` block, so these considerations need to be balanced.

Reply:
👍🏽